src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java

```diff
@@ -51,23 +57,48 @@ public class DynamicSimulationServiceImpl implements DynamicSimulationService {
 
     private final DynamicSimulationClient dynamicSimulationClient;
 
+    private final NetworkService networkService;
+
     private final NetworkModificationTreeService networkModificationTreeService;
 
     public DynamicSimulationServiceImpl(ObjectMapper objectMapper,
                                         DynamicMappingClient dynamicMappingClient,
                                         TimeSeriesClient timeSeriesClient,
                                         DynamicSimulationClient dynamicSimulationClient,
+                                        NetworkService networkService,
                                         NetworkModificationTreeService networkModificationTreeService) {
         this.objectMapper = objectMapper;
         this.dynamicMappingClient = dynamicMappingClient;
         this.timeSeriesClient = timeSeriesClient;
         this.dynamicSimulationClient = dynamicSimulationClient;
+        this.networkService = networkService;
         this.networkModificationTreeService = networkModificationTreeService;
     }
 
+    private String getVariantId(UUID nodeUuid) {
+        return networkModificationTreeService.getVariantId(nodeUuid);
+    }
+
+    private UUID getReportUuid(UUID nodeUuid) {
+        return networkModificationTreeService.getReportUuid(nodeUuid);
+    }
```

Comment:
Why do you create those private accessors instead of juste directly calling the functions ?